### PR TITLE
Convert filter partial to a view component

### PR DESCRIPTION
**Pull request by gsmendoza**
## Goal

As a Solidus Starter Frontend user
I want the filter partial to be converted to a view component
So that it would be easier to maintain and test

## How Has This Been Tested?

* Manually tested pages, primarily
  * http://localhost:3000/
  * http://localhost:3000/t/categories/clothing
* Added test for FilterComponent.

## Demo

https://www.loom.com/share/a5fdfce57ab647fda61182ce5bb3d274

## Types of changes

- N/A: Bug fix (non-breaking change which fixes an issue)
- N/A: New feature (non-breaking change which adds functionality)
- N/A: Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- N/A: My change requires a change to the documentation.
- N/A: I have updated the documentation accordingly.

### CODE REVIEW

File: templates/app/components/filter_component.rb [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        value,
+        filter_list_item_checked?(value),
+        id: id)
+
+      concat label_tag(id, name)
+    end
+  end
+
+  def filter_list_item_id(name)
+    sanitize_to_id("#{filter[:name]}_#{name}")
```
Comment (gsmendoza) [inline templates/app/components/filter_component.rb:48]: We're using `sanitize_to_id` to ensure that the check box ID matches the `for` attribute of the item label tag.

**Review by kennyadsl — APPROVED** [review]
Thanks @gsmendoza, looks great. Did you check if we can remove some feature specs now that we have the component behavior tested in isolation?

File: templates/app/components/filter_component.rb [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    id = filter_list_item_id(name)
+
+    content_tag :li do
+      concat check_box_tag(
+        "search[#{filter[:scope].to_s}][]",
+        value,
+        filter_list_item_checked?(value),
+        id: id)
+
+      concat label_tag(id, name)
```
[the remark this answers is not in the document]
Comment (gsmendoza) [reply templates/app/components/filter_component.rb:43]: @kennyadsl FYI. Sorry, I forgot to post these comments before submitting the PR for review.

**Comment by gsmendoza** [comment]
@kennyadsl 

> Did you check if we can remove some feature specs now that we have the component behavior tested in isolation?

Oh, I didn't check the feature specs. I'll create a ticke to check if we have feature specs that test behavior already tested by component specs.

**Pull request merged**